components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -3248,6 +3247,60 @@ public boolean isSubscribed(APIIdentifier apiIdentifier, String userId) throws A
+        try (Connection connection = APIMgtDBUtil.getConnection();
+            PreparedStatement ps =
+                    connection.prepareStatement(SQLConstants.GET_APP_API_PRODUCT_USAGE_BY_PROVIDER_SQL)) {
+            ps.setString(1, APIUtil.replaceEmailDomainBack(providerName));
+            try (ResultSet result = ps.executeQuery()) {
+                Map<String, UserApplicationAPIUsage> userApplicationUsages = new TreeMap<String, UserApplicationAPIUsage>();
+                while (result.next()) {
+                    int subId = result.getInt("SUBSCRIPTION_ID");
+                    Map<String, String> keyData = getAccessTokenData(subId);
+                    String accessToken = keyData.get("token");
+                    String tokenStatus = keyData.get("status");
+                    String userId = result.getString("USER_ID");
+                    String application = result.getString("APPNAME");
+                    int appId = result.getInt("APPLICATION_ID");
+                    String subStatus = result.getString("SUB_STATUS");
+                    String subsCreateState = result.getString("SUBS_CREATE_STATE");
+                    String key = userId + "::" + application;
+                    UserApplicationAPIUsage usage = userApplicationUsages.get(key);
+                    if (usage == null) {
+                        usage = new UserApplicationAPIUsage();
+                        usage.setUserId(userId);
+                        usage.setApplicationName(application);
+                        usage.setAppId(appId);
+                        usage.setAccessToken(accessToken);
+                        usage.setAccessTokenStatus(tokenStatus);
+                        userApplicationUsages.put(key, usage);
+                    }
+                    APIProductIdentifier apiProductId = new APIProductIdentifier(result.getString("API_PRODUCT_PROVIDER"), result.getString
+                            ("API_PRODUCT_NAME"), result.getString("API_PRODUCT_VERSION"));
+                    SubscribedAPI apiSubscription = new SubscribedAPI(new Subscriber(userId), apiProductId);
+                    apiSubscription.setSubStatus(subStatus);
+                    apiSubscription.setSubCreatedStatus(subsCreateState);
+                    apiSubscription.setUUID(result.getString("SUB_UUID"));
+                    apiSubscription.setTier(new Tier(result.getString("SUB_TIER_ID")));
+                    Application applicationObj = new Application(result.getString("APP_UUID"));
+                    apiSubscription.setApplication(applicationObj);
+                    usage.addApiSubscriptions(apiSubscription);
+                }
+                return userApplicationUsages.values().toArray(new UserApplicationAPIUsage[userApplicationUsages.size()]);
+            }
+        } catch (SQLException e) {
+            handleException("Failed to find API Product Usage for :" + providerName, e);
```

Comment:
Will this be a client error or server error?

Reply:
This will be a server side error(classified as 500). All SQLExceptions will be considered in this way.